libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.ts

```diff
@@ -40,83 +46,96 @@ export class ProductSwitchBodyComponent implements OnInit {
-    listMode: boolean;
+    constructor(@Optional() private readonly _rtlService: RtlService) {}
 
     /** @hidden */
     ngOnInit(): void {
+        this._subscriptions.add(this._rtlService?.rtl.subscribe((isRtl) => (this._isRtl = isRtl)));
+
+        this._checkSize();
+    }
+
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** @hidden */
+    @HostListener('window:resize', [])
+    onResize(): void {
         this._checkSize();
     }
 
-    /**
-     * @hidden
-     */
-    itemClick(item: ProductSwitchItem, event: any): void {
+    /** @hidden */
+    _itemClick(item: ProductSwitchItem, event: MouseEvent): void {
+        this.products.forEach((product) => (product.selected = product === item));
+
         this.itemClicked.emit();
         if (item.callback) {
             item.callback(event);
         }
     }
 
     /** Method called on products change */
-    productSwitchItemsChangeHandle(dropEvent: FdDropEvent<ProductSwitchItem>): void {
+    _productSwitchItemsChangeHandle(dropEvent: FdDropEvent<ProductSwitchItem>): void {
         this.productsChange.emit(dropEvent.items);
         this.products = dropEvent.items;
     }
 
     /** @hidden */
-    @HostListener('window:resize', [])
-    onResize(): void {
-        this._checkSize();
-    }
-
-    /** @hidden */
-    keyDownHandle(event: KeyboardEvent): void {
+    _keyDownHandle(event: KeyboardEvent): void {
         const target = <HTMLElement>event.target;
-        const i = Array.from(target.parentElement.children).indexOf(target);
+
         if (!KeyUtil.isKeyCode(event, TAB)) {
             event.preventDefault();
         }
+
         if (KeyUtil.isKeyCode(event, [ENTER, SPACE])) {
             target.click();
-        } else if (!this.isListMode()) {
+        } else if (!this._isListMode()) {
+            const i = Array.from(target.parentElement.children).indexOf(target);
             this._handleNoListKeydown(event, target, i);
-        } else if (this.isListMode() && KeyUtil.isKeyCode(event, [DOWN_ARROW, UP_ARROW])) {
+        } else if (this._isListMode() && KeyUtil.isKeyCode(event, [DOWN_ARROW, UP_ARROW])) {
             this._handleListArrowUpDown(event, target);
         }
     }
 
     /** @hidden */
-    public isSmallMode(): boolean {
-        return this.products && this.products.length < 7;
+    _isSmallMode(): boolean {
+        return this.products?.length < 7;
     }
 
     /** @hidden */
-    public isListMode(): boolean {
-        return this.listMode || this.forceListMode;
+    _isListMode(): boolean {
+        return this._istMode || this.forceListMode;
     }
 
     /** @hidden */
     private _checkSize(): void {
-        if (this.isSmallMode()) {
-            this.listMode = window.innerWidth < 588;
+        if (this._isSmallMode()) {
+            this._istMode = window.innerWidth < 588;
         } else {
-            this.listMode = window.innerWidth < 776;
+            this._istMode = window.innerWidth < 776;
```

Comment:
It would be better to create constants for all "magic" numbers. Because I don't understand what these numbers mean

Reply:
I understand that it was not you who wrote it, but it would be nice to fix it for the developers who will work after you.